include/fiction/algorithms/simulation/sidb/quicksim.hpp

```diff
@@ -118,30 +135,60 @@ void quicksim(const Lyt& lyt, const quicksim_params& ps = quicksim_params{}, qui
             st.valid_lyts.push_back(charge_distribution_surface<Lyt>{charge_lyt});
         }
 
-        auto       best_energy = std::numeric_limits<double>::max();
-        const auto bound       = static_cast<uint64_t>(std::round(0.6 * static_cast<double>(charge_lyt.num_cells())));
-        for (uint64_t z = 0u; z < ps.interation_steps; z++)
+        // If the number of threads is initially set to zero, the simulation is run with one thread.
+        uint64_t num_threads = std::max(ps.number_threads, uint64_t{1});
+
+        // split the iterations among threads
+        const auto iter_per_thread =
+            std::max(ps.interation_steps / num_threads,
+                     uint64_t{1});  // If the number of set threads is greater than the number of iterations, the
+                                    // number of threads defines how many times QuickSim is repeated
+
+        const auto bound = static_cast<uint64_t>(std::round(0.6 * static_cast<double>(charge_lyt.num_cells())));
```

Comment:
Can you please provide a description via comment of this line of code? Is it an upper or lower bound? What does the `0.6` multiplier do?